.github/workflows/build-packages.yml

```diff
@@ -64,41 +72,50 @@ jobs:
       - name: Upload artifact
         uses: actions/upload-artifact@v2
         with:
-          name: bob-rpm
+          name: bob-rpm-${{ matrix.bob_key_size }}
           path: rpmbuild/RPMS/x86_64/bob-${{ env.VERSION }}-${{ github.run_id }}.x86_64.rpm
 
   upload-rpm:
     runs-on: ubuntu-latest
     needs: build-rpm
     environment: prod
+    strategy:
+      matrix:
+        bob_key_size: [8, 16]
+        include:
+          - bob_key_size: 16
+            asset_name_postfix: -key16
```

Comment:
You can use conditional step that generates this name and save it to environment variable only when bob_key_size != 8